nebula_hw_interfaces/src/nebula_hesai_hw_interfaces/hesai_hw_interface.cpp

```diff
@@ -27,17 +28,21 @@
 
 #include <boost/asio.hpp>
 
+#include <utility>
+
 namespace nebula::drivers
 {
 
 using std::string_literals::operator""s;
 using nlohmann::json;
 
-HesaiHwInterface::HesaiHwInterface()
-: cloud_io_context_{new ::drivers::common::IoContext(1)},
+HesaiHwInterface::HesaiHwInterface(std::shared_ptr<loggers::Logger> logger)
+: logger_(std::move(logger)),
```

Comment:
is this move not misleading?

Reply:
While passing a `shared_ptr` by value and then move-constructing a member from can be more efficient ([discussion here](https://stackoverflow.com/a/12002668)), I agree that clarity takes precedence here.

Made `const&` without `std::move` in 755d722.